src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -3,61 +3,223 @@
     :copyright: (c) 2008-2023 Sigasi
     :license: BSD, see LICENSE for more details.
 """
-import os
 import argparse
 import pathlib
 
+from SigasiProjectCreator import VerilogVersion, VhdlVersion
+
 
 class ArgsAndFileParser:
     options = None
 
-    def __init__(self, usage):
+    def __init__(self):
+        ArgsAndFileParser.options = None
         self.parser = argparse.ArgumentParser(prog='SigasiProjectCreator')
```

Comment:
`self.parser` should not be a object variable of ArgsAndFileParser since it should be changed after initialization. Most often, they are used as single-use objects:

```python
def parse_arguments(*args):
    parser = argparse.ArgumentParser(...)
    parser.add_argument(...)
    return parser.parse_args(args)
```

Reply:
fixed: parsing is now part of the constructor